### DISCUSSION TITLE
messages between nodes,how to do?

### DISCUSSION THREAD

**Discussion by jiait** [ergo-services/ergo#164] [General]
Can GO native structure data be directly message-sent between nodes? Can I call etf.RegisterType registration to decode messages?

**Comment by halturin** [comment]
>etf.RegisterType

That is exactly was created for :). But you should register type on both sides - sender node and receiver.

**Comment by jiait** [comment]
How does this serialization compare to protobuf or msgpack serialization?

**Comment by jiait** [comment]
Etfs.RegisterType Currently some structure data can not be converted, such as map[string]*value,map value is pointer mode can not be used? Is not fully serialized JSON, this way there is no problem at all?

**Comment by jiait** [comment]
type NodeMsg struct {
	Attach  map[string]string
	Bin     []byte
}
Why is map not supported？

**Reply by halturin** [reply]
It must be declared as a type and registered

**Reply by jiait** [reply]
How to register? Ask for help

**Reply by halturin** [reply]
type Attach map[string]string
type NodeMsg struct {
  Attach Attach
  Bin []byte
}

... 
etf.RegisterType(...) // register Attach type
etf.RegisterType(...) // register NodeMsg

**Reply by jiait** [reply]
Okay, got it. Thank you. Thank you
